index.js

```diff
@@ -177,21 +181,71 @@ program
 */
 
 
-var ercChecker = function (originalHTML, reproducedHTML, outputPath, checkID, createParentDirectoriesInOutputPath, silenceDebuggers, saveMetadata) {
+var ercChecker = function (originalHTML, reproducedHTML, checkID, outputPath, saveDiffHTML, saveMetadataJSON, createParentDirectoriesInOutputPath, silenceDebuggers) {
 
+	var checkId = '';
 	return new Promise( function (resolve, reject) {
 
-
-
 		var pathOriginalHTML = originalHTML,
 			pathReproducedHTML = reproducedHTML,
 			brokenPath = false,
 			checkStart = Date.now();
 
-		var outputName = outputPath;
+		if ((saveDiffHTML || saveMetadataJSON) && outputPath == null) {
+			debugERROR("Cannot save files without output path.".red);
+			reject(new Metadata(checkStart, new Error("Requested saving check files without specifying an output path.")));
+		}
+
+		debug(checkID);
+
 		if (checkID) {
-			var checkIDString = '_'+checkID;
+			checkId = '_'+checkID;
+		}else {
+			checkId = '';
 		}
+		var fileOutputPath;
+		try {
+			if (outputPath) {
+				fileOutputPath = fs.realpathSync(outputPath);
+			}
+		}
+		catch (e) {
+			if (createParentDirectoriesInOutputPath) {
+
+				if (process.platform === 'win32') {
+					exec("mkdir " + fileOutputPath,
+						function (err) {
+							if (err) {
+								debugERROR("Could not create output directory.".red);
+								reject( new Metadata(checkStart, err) );
+							}
+						}
+					);
+				}
+				else {
+					exec("mkdir -p " + fileOutputPath,
+						function (err) {
+							if (err) {
+								debugERROR("Could not create output directory.".red);
+								reject( new Metadata(checkStart, err) );
+							}
+						}
+					);
+				}
+				fileOutputPath = fs.realpathSync(outputPath);
+			}
+			else {
+				debugERROR("Output path non-existent, option to create it was not set. ".red);
```

Comment:
You should at least log `e` - or can "output path does not exist" be the only error happening here?